cpu/atmega_common/periph/Kconfig

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2021 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+if TEST_KCONFIG && CPU_COMMON_ATMEGA
+
+config MODULE_ATMEGA_COMMON_PERIPH
+    bool
+    default y
+    help
+        Common ATmega peripheral drivers.
+
+config MODULE_ATMEGA_PCINT
+    bool
+    depends on HAS_ATMEGA_PCINT0
+
+config MODULE_ATMEGA_PCINT0
+    bool
+    depends on HAS_ATMEGA_PCINT0
+    default MODULE_ATMEGA_PCINT
```

Comment:
```suggestion
```

I might just not understand Kconfig enough, but I think that should be dropped.

Reply:
This is trying to achieve the behaviour expressed in 
https://github.com/RIOT-OS/RIOT/blob/5a9c659f5df5788faff0ae5aea23697e004132b3/cpu/atmega_common/Makefile.dep#L7-L10

So if `MODULE_ATMEGA_PCINT` and `HAS_ATMEGA_PCINT0` are enabled, `MODULE_ATMEGA_PCINT0` is enabled. Is that correct?

Reply:
OK, I get it now. But this should be added to MODULE_ATMEGA_PCINT1, 2, and 3 as well.

Reply:
I can do so, but:
1. it would automatically enable those banks if CPUs have them, and I understood from the doc that it is up to the boards to say so.
2. I don't think it matches the behaviour of Makefile.dep, as for those it only adds the FEATURE_REQUIRED when the specific bank is required.

I can change also the Makefile.dep to match that if it's the expected one.

Reply:
That's strange. When the PR landed, the behavior was: If you select a specific bank, only that gets used. If you select `atmega_pcint`, all available banks get added.

Reply:
It still is like this. The issue is that only PCINT bank 0 is common to all ATmega MCUs. So the implementation is done at CPU level, not in atmega_common.

Reply:
Hmm ok I thought it was up to the boards as the doc is there https://github.com/RIOT-OS/RIOT/blob/master/boards/common/atmega/doc.txt.

I will add the default to all of them.

Reply:
It would indeed make sense to move that documentation to `cpu/atmega_common` instead.

Reply:
Done